v2/api/devices/customizations/iot_hub_extension.go

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package customizations
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/iothub/armiothub"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	storage "github.com/Azure/azure-service-operator/v2/api/devices/v1api20210702storage"
```

Comment:
minor: call this `iothub` or something? Calling it storage makes me think Azure storage.